### ISSUE TITLE
Wrong `whitespace/braces` warnings for brace initialization

### ISSUE THREAD

**Issue by chen3feng** [cpplint/cpplint#174]
xxx.cpp

```cpp
int main() {
  static thread_local std::mt19937 generator(std::random_device{}());
  result = std::string{"hello"} + ex.what();
}
```

```console
xxx.cpp:0:  No copyright message found.  You should have a line: "Copyright [year] <Copyright Owner>"  [legal/copyright] [5]
xxx.cpp:2:  Missing space before {  [whitespace/braces] [5]
xxx.cpp:3:  Missing space before {  [whitespace/braces] [5]
Done processing xxx.cpp
Total errors found: 3
```
